### PR TITLE
chore(infra): Add list of attributes that can't be removed due to emptiness

### PR DESCRIPTION
**Pull request by AndesKrrrrrrrrrrr**
Fixes an issue in https://github.com/island-is/island.is/pull/16832, where `service:api` would fail on startup due to missing `SERVERSIDE_FEATURES_ON`

## Release Notes

- **New Features**
	- Introduced a new environment variable `SERVERSIDE_FEATURES_ON` across multiple services, allowing for enhanced configuration options.
  
- **Impact**
	- This variable is initialized with an empty string and does not alter existing functionalities, but it may serve as a feature toggle for future enhancements.

### CODE REVIEW

File: charts/services/xroad-collector/values.prod.yaml [hunk 1]
```diff
@@ -18,7 +18,6 @@ service:
     LOG_LEVEL: 'info'
     NODE_OPTIONS: '--max-old-space-size=230 -r dd-trace/init'
     NODE_TLS_REJECT_UNAUTHORIZED: '0'
-    SERVERSIDE_FEATURES_ON: 'driving-license-use-v1-endpoint-for-v2-comms'
```
Comment (eirikurn) [inline charts/services/xroad-collector/values.prod.yaml:21]: Are you intentionally removing this serverside feature?

File: infra/src/cli/generate-chart-values.ts [hunk 2]
```diff
@@ -32,7 +34,9 @@ const removeEmptyStringProperties = (obj: any): any => {
 
   return Object.fromEntries(
     Object.entries(obj)
-      .filter(([_, value]) => value !== '') // Filter out empty strings
+      .filter(
+        ([key, value]) => value !== '' && !NON_EMPTYABLE_PROPERTIES.has(key),
+      ) // Filter out empty strings
```
Comment (eirikurn) [inline infra/src/cli/generate-chart-values.ts:39]: Think this condition might be wrong since it is removing non-empty values for SERVERSIDE_FEATURES_ON.

Comment (AndesKrrrrrrrrrrr) [reply infra/src/cli/generate-chart-values.ts]: Updated the condition, fixed :heart:

**Review by busla — APPROVED** [review]
LGTM

**Review by eirikurn — APPROVED** [review]

**Pull request merged**